compiler/AST/ParamForLoop.cpp

```diff
@@ -478,6 +479,26 @@ CallExpr* ParamForLoop::foldForResolve()
   return noop;
 }
 
+// Checks things like bounding and paramness of the range.
+// The calls to chpl_compute_low_param_loop_bound and
+// chpl_compute_high_param_loop_bound in buildParamForLoop should ensure that
+// the appropriate type is used for these loops
+void ParamForLoop::validateLoop(VarSymbol* lvar,
+                                VarSymbol* hvar,
+                                VarSymbol* svar) {
+  if (!lvar            || !hvar            || !svar)
+    USR_FATAL(this,
+              "param for loop must be defined over a bounded param range");
+
+  if (!lvar->immediate || !hvar->immediate || !svar->immediate)
+    USR_FATAL(this,
+              "param for loop must be defined over a bounded param range");
```

Comment:
Is this message ever triggered within the compiler?  I'd expect that it would always catch in the module code first?  (which seems preferable).  Oh, maybe only if the stride is not a param?

Reply:
You're right, this will probably only trigger for the stride (or if we accidentally remove the param return intent from those functions, or stop replacing things correctly in the compiler).

Reply:
How about switch this to INT_ASSERT ? Because it is not user fault that we made a mistake in the module code or the compiler.

Reply:
Same for the int check below, as long as we do catch it in the module code.

Reply:
The stride not being a param is on the user, though.